test/jest_integration/saml_auth.test.ts

```diff
@@ -0,0 +1,334 @@
+    const driver = getDriver().build();
+    await driver.manage().deleteAllCookies();
+    await driver.get(urlWithHash);
+    await driver.wait(until.elementsLocated(By.xpath('/html/body/nav/div/div[1]/a/i/span')), 60000);
+    await driver.findElement(By.xpath(signInBtnXPath)).click();
+    await driver.wait(
+      until.elementsLocated(By.xpath('/html/body/div[1]/div/header/div/div[2]')),
+      20000
+    );
+    const windowHash = await driver.getCurrentUrl();
+    expect(windowHash).toEqual(urlWithHash);
+    const cookie = await driver.manage().getCookies();
+    expect(cookie.length).toEqual(2);
+    await driver.manage().deleteAllCookies();
+    await driver.quit();
+  });
+
+  it('Tenancy persisted after Logout in SAML', async () => {
+    const driver = getDriver().build();
+
+    await driver.get('http://localhost:5601/app/opensearch_dashboards_overview#/');
+
+    await driver.findElement(By.xpath(signInBtnXPath)).click();
+
+    await driver.wait(until.elementsLocated(By.xpath(pageTitleXPath)), 10000);
+
+    await driver.wait(
+      until.elementsLocated(By.xpath('//button[@aria-label="Closes this modal window"]')),
+      10000
+    );
+
+    // Select Global Tenant Radio Button
+    const radio = await driver.findElement(By.xpath('//input[@id="global"]'));
+    await driver.executeScript('arguments[0].scrollIntoView(true);', radio);
+    await driver.executeScript('arguments[0].click();', radio);
+
+    await driver.findElement(By.xpath('//button[@data-test-subj="confirm"]')).click();
+
+    await driver.wait(until.elementsLocated(By.xpath(userIconBtnXPath)), 10000);
+
+    await driver.findElement(By.xpath(userIconBtnXPath)).click();
+
+    await driver.findElement(By.xpath('//*[@data-test-subj="log-out-1"]')).click();
+
+    // RELOGIN AND CHECK TENANT
+
+    await driver.wait(until.elementsLocated(By.xpath(signInBtnXPath)), 10000);
+
+    await driver.findElement(By.xpath(signInBtnXPath)).click();
+
+    await driver.wait(until.elementsLocated(By.xpath(skipWelcomeBtnXPath)), 10000);
+
+    await driver.findElement(By.xpath(skipWelcomeBtnXPath)).click();
+
+    await driver.findElement(By.xpath(userIconBtnXPath)).click();
+
+    await driver.wait(until.elementsLocated(By.xpath(tenantNameLabelXPath)), 10000);
+
+    const tenantName = await driver.findElement(By.xpath(tenantNameLabelXPath)).getText();
+
+    await driver.manage().deleteAllCookies();
+    await driver.quit();
+  });
+});
+
+function getDriver() {
+  return new Builder().forBrowser('firefox').setFirefoxOptions(new Options().headless());
```

Comment:
Would it make sense at all to parameterize `firefox` to be able to run these tests with a different browser?